rep-2014.rst

```diff
@@ -0,0 +1,232 @@
+
+
+- `ros2_benchmarking <https://github.com/piappl/ros2_benchmarking/>`_ : First implementation available for ROS 2, aimed to provide a framework to compare ROS and ROS 2 communications.
+- `performance_test <https://gitlab.com/ApexAI/performance_test/>`_: Tool is designed to measure inter and intra-process communications. Runs at least one publisher and at least one subscriber, each one in one independent thread or process and records different performance metrics. It also provides a way to generate a report with the results through a different package.
+- `ros2-performance <https://github.com/irobot-ros/ros2-performance/>`_: Another framework to evaluate ROS communications and inspired on `performance_test`. There's a decent rationale in the form of a proposal, a good evaluation of prior work and a well documented set of experiments.
+- `system_metrics_collector <https://github.com/ros-tooling/system_metrics_collector/>`_: A lightweight and *real-time* metrics collector for ROS 2. Automatically collects and aggregates *CPU* % used and *memory* % performance metrics used by both system and ROS 2 processes. Data is aggregated in order to provide constant time average, min, max, sample count, and standard deviation values for each collected metric. *Deprecated*.
+- `ros2_latency_evaluation <https://github.com/Barkhausen-Institut/ros2_latency_evaluation/>`_: A tool to benchmarking performance of a ROS 2 Node system in separate processses (initially focused on both inter-process and intra-process interactions, later focused). Forked from `ros2-performance`.
+- `ros2_timer_latency_measurement <https://github.com/hsgwa/ros2_timer_latency_measurement/>`_:  A minimal *real-time safe* testing utility for measuring jitter and latency.  Measures nanosleep latency between ROS child threads and latency of timer callbacks (also within ROS) across two different Linux kernel setups (`vanilla` and a `RT_PREEMPT`` patched kernel).
+- `buildfarm_perf_tests <https://github.com/ros2/buildfarm_perf_tests/>`_: Tests which run regularly on the official ROS 2 buildfarm. Formally, extends `performance_test` with additional tests that measure additional metrics including CPU usage, memory, resident anonymous memory or virtual memory.
+- `ros-tracing <https://gitlab.com/ros-tracing/>`_: Tracing tools for ROS 2 built upon LTTng which allow collecting runtime execution information on real-time distributed systems, using the low-overhead LTTng tracer. Performance evaluation can be scripted out of the data collected from all these trace points. The ROS 2 core layers (rmw, rcl, rclcpp) have been instrumented with LTTng probes which allow collecting information of ROS 2 targets without the need to modify the ROS 2 core code (*system under test)*. There various publications available about *ros-tracing*  [3]_ [4]_ and it is used actively to benchmark ROS 2 in real scenarios including perception and mapping [5]_, hardware acceleration [6]_ [7]_ or self-driving mobility [8]_.
```

Comment:
@vmayoral FYI `ros2_tracing` moved to GitHub: https://github.com/ros2/ros2_tracing

Reply:
@christophebedard thanks a lot. See 4b98c34e50b0599bf7bc26206ced5a722acc1d5c. Feel free to send a suggestion if you spot anything like this and/or have any suggestion. I'd love to have your views on this and wouldn't mind co-authoring it together.